### ISSUE TITLE
Crisp is not crisp enough

### ISSUE THREAD

**Issue by djmittens** [photonstorm/phaser#1586]
http://examples.phaser.io/_site/view_full.html?d=display&f=render+crisp.js&t=render%20crisp

If you zoom really close in the crisp example you will see that the pixels are not really crisp, each pixel is more like a rounded square than a sharp square, is there a better possible fix for this?

Also in case it helps, im using chrome browser on yosemite.

**Comment by wayfu** [comment]
The browser's zoom isn't crisp, it's not a phaser issue. If you take a screenshot and zoom in on it in an image manipulation software, you will see that the pixels are perfectly crisp.

**Comment by djmittens** [comment]
This is not what im talking about. When you use the zoom in button on the example, not browsers zoom, the picture looses crispness.  However by following this tutorial you can stretch your canvas one way or another and the crispness stays, however this way of fixing it is very hacky and doesnt work with webgl thats why i was wondering maybe there is a much better solution, because its been a long time since that post.

http://www.photonstorm.com/phaser/pixel-perfect-scaling-a-phaser-game

**Comment by gingerbeardman** [comment]
are you doing

```sprite.smoothed = false;```

See:
http://examples.phaser.io/_site/view_full.html?d=arcade%20physics&f=body+scale.js&t=body%20scale

**Comment by djmittens** [comment]
I am not.  Is there no way to do it without doing that? i mean seriously lame if you want to scale the whole game and you have to do it sprite by sprite especially in large levels where map loading happens outside my control.

The article i linked to is definately a step in the right direction and i feel like its something that should be supported by the game engine itself, it seems to work with both canvas and webgl, with the only drawback of mouse input, if i cant figure out a native way of doing it then i will probably have to go ahead and just figure out the mouse clicks.  Unfortunately copying the whole canvas like that is also resource intensive so im not even sure how thats going to impact my game once im ready to release it to mobile devices.

**Comment by photonstorm** [comment]
*Phaser.Canvas.setImageRenderingCrisp(game.canvas)*

This now works perfectly in Canary (and I believe Chrome stable, although test for yourself). It works regardless of Canvas or WebGL mode and looks fantastic. Sadly, most other browsers ignore it, although being part of the spec that should change.

*PIXI.scaleModes.DEFAULT = PIXI.scaleModes.NEAREST*

This tells WebGL to use nearest neighbour scaling, which again gives lovely crisp pixel art (regardless of what the browser supports). Only works in WebGL though.

*Phaser.Canvas.setSmoothingEnabled(game.context, false)*

Disables image smoothing on the context itself, so solves the issue caused above in Canvas mode. You can use this in conjunction with the above too.

None of these (except the top one) solve the problem of scaling the canvas with CSS though.

In short, there is no 100% guaranteed way to do it yet that is perfectly consistent across all current browsers. 'pixelate' is the future, but doesn't handle the past.

**Comment by djmittens** [comment]
So i tried these methods out and i am using a MAC chrome stable, the first 2 do not work, with web gl ( which is what i care about).

The chrome version i am using is
 
Version 40.0.2214.115 (64-bit)
 
Google Chrome is up to date.
